sw/device/tests/aon_timer_irq_test.c

```diff
@@ -123,27 +123,11 @@ static void execute_test(dif_aon_timer_t *aon_timer, uint64_t irq_time_us,
   }
   // Capture the current tick to measure the time the IRQ will take.
   uint32_t start_tick = (uint32_t)tick_count_get();
-  uint32_t time_elapsed = 0;
-
-  // Disable interrupts to be certain interrupt doesn't occur between while
-  // condition check and `wait_for_interrupt` (so WFI misses that interrupt).
-  irq_global_ctrl(false);
-
-  // Only enter WFI loop if we haven't already seen the interrupt.
-  if (peripheral != kTopEarlgreyPlicPeripheralAonTimerAon) {
-    do {
-      wait_for_interrupt();
-      // WFI ignores global interrupt enable, so enable it now and then
-      // immediately disable it. If there is an interrupt pending it will be
-      // taken here between the enable and disable. This confines the interrupt
-      // to a known place avoiding missed wakeup issues.
-      irq_global_ctrl(true);
-      irq_global_ctrl(false);
-      time_elapsed = (uint32_t)irq_tick - start_tick;
-    } while (peripheral != kTopEarlgreyPlicPeripheralAonTimerAon &&
-             time_elapsed < sleep_range_h);
-  }
 
+  ATOMIC_WAIT_FOR_INTERRUPT(peripheral ==
```

Comment:
This conversion slightly changes the behaviour: the above loop would stop after `sleep_range_h` but the new one does not. I don't know if that changes the anything for the test?

Reply:
You are right, now the test will relying on the bazel timeout just like many other tests. When I wrote this test we were using mason, so a timeout control was required.